src/TLuaInterpreter.cpp

```diff
@@ -9885,63 +9874,56 @@ int TLuaInterpreter::Echo(lua_State* L)
 // Documentation: https://wiki.mudlet.org/w/Manual:Lua_Functions#echoPopup
 int TLuaInterpreter::echoPopup(lua_State* L)
 {
-    string a1 = "";
-    string a2;
-    QStringList _hintList;
-    QStringList _commandList;
+    QString window;
+    QString text;
+    QStringList hintList;
+    QStringList commandList;
     bool customFormat = false;
     int s = 1;
     int n = lua_gettop(L);
     // console name is an optional first argument
     if (n > 4) {
         if (!lua_isstring(L, s)) {
-            lua_pushstring(L, "echoPopup: wrong argument type");
-            lua_error(L);
-            return 1;
+            lua_pushfstring(L, "echoPopup: bad argument #%d type (window name as string expected, got %s!)", s, luaL_typename(L, s));
```

Comment:
Isn't this optional (can be `main` or *not supplied* to send to main console window).

Reply:
Yes I think it is